src/ui/public/vislib/lib/axis.js

```diff
@@ -21,9 +23,13 @@ export default function AxisFactory(Private) {
       this.el = args.el;
       this.xValues = args.xValues;
       this.ordered = args.ordered;
-      this.xAxisFormatter = args.xAxisFormatter;
+      this.axisFormatter = args.type === 'category' ? args.xAxisFormatter : args.yAxisFormatter;
       this.expandLastBucket = args.expandLastBucket == null ? true : args.expandLastBucket;
       this._attr = _.defaults(args._attr || {});
+      this.scale = null;
+      this.domain = [args.yMin, args.yMax];
```

Comment:
its only used for yAxis ... the code in this PR is pretty bad as i only joined the 2 files to make it clear how i progressed. yAxis and xAxis have many functions that actually do same/similar thing but are named differently and sometimes work in different ways (like on one axis you are supposed to pass a scale to function which returns domain and on the other one you are supposed to pass in limits and then apply the return value on scale ... i wouldn't worry about all that in this PR ... its just to show steps of progress ... so putting all the functions from both files into one ... then we can clearly see in the diffs what have been changed from here on ...